src/base/Notifier.sol

```diff
@@ -65,13 +66,17 @@ abstract contract Notifier is INotifier {
     function _unsubscribe(uint256 tokenId, PositionConfig calldata config) internal {
         _positionConfigs(tokenId).setUnsubscribe();
         ISubscriber _subscriber = subscriber[tokenId];
+        if (_subscriber == NO_SUBSCRIBER) NotSubscribed.selector.revertWith();
```

Comment:
i see that this check makes sense, surprised it wasnt here before
doesnt one of these issues say to add it? i cant find it

Reply:
I think it may make more sense to check _postionConfigs().hasSubscriber? i guess see if its cheaper in the case where they do have a subscriber?

Reply:
yep looked like .hasSubscriber() was slightly cheaper! refactored it